CMakeLists.txt

```diff
@@ -162,8 +168,9 @@ set(CMAKE_XCODE_ATTRIBUTE_MARKETING_VERSION ${APP_SHORT_VERSION})
 set(CMAKE_XCODE_ATTRIBUTE_ONLY_ACTIVE_ARCH NO)
 set(CMAKE_XCODE_ATTRIBUTE_ONLY_ACTIVE_ARCH[variant=Debug] YES)
 
-if(BUILD_SINGLE_APP)
-	add_compile_definitions(SINGLE_PROCESS_APP=1)
+if(ENABLE_SINGLE_APP_BUILD)
+	add_definitions(-DSINGLE_PROCESS_APP=1)
+	#add_compile_definitions(SINGLE_PROCESS_APP=1) # requires cmake 3.12
```

Comment:
Do we need this line anymore?

Reply:
Which one?
`SINGLE_PROCESS_APP=1` define is used in c++ code
commented-out `add_compile_definitions` is likely to be needed in future - looks like CMake is replacing `add_definitions` with a set of add_X commands, so `add_definitions`  is likely be removed/deprecated at some point

Reply:
Why can not we use add_compile_definition right now?

Reply:
I think we already depend on cmake 3.16 because we use its builtin precompiled headers instead of cotire

Reply:
But if it is still compilable with cmake 3.10 I suggest to remove commented out code.

Reply:
Precompiler headers check for CMake version, so they work even with older cmake.
I guess I remove commented-out code from here, at least for now.

A separate question is ancient MXE build which uses (among other things) CMake 3.10. But this is definitely out of scope of this PR, will write this on Slack